### ISSUE TITLE
BatchRenormalization uses Numpy code where it should use TF code

### ISSUE THREAD

**Issue by waleedka** [farizrahman4u/keras-contrib#127]
Looking at the code of [BatchRenormalization ](https://github.com/farizrahman4u/keras-contrib/blob/master/keras_contrib/layers/normalization.py), I think there is a serious issue there. This section, for example, inside the call() function:
```
# update r_max and d_max
t_val = K.get_value(self.t)
r_val = self.r_max_value / (1 + (self.r_max_value - 1) * np.exp(-t_val))
d_val = self.d_max_value / (1 + ((self.d_max_value / 1e-3) - 1) * np.exp(-(2 * t_val)))
t_val += float(self.t_delta)
```

The main issue is that the code uses Python/Numpy operations, rather than Keras/TensorFlow operations. Note that the layer call() function is executed only once when the model is created. It does *not* get called repeatedly for every batch. The call() function should create a TF/Theano computation graph rather than do the computation directly in Python/Numpy.

What's happening there is that it calls K.get_value(), which internally creates a TF session and runs it to get the float value of ```self.t```. Then, r_val and d_val and calculated in Python code and used in subsequent update ops. Since r_val and d_val are computed only once when the model is created, they'll have values of 1 and 0 respectively and will remain so. In other words, if I'm reading this correctly, this layer is operating as a regular batch normalization layer, not as a batch re-normalization layer. 

I didn't run the code to confirm so I could be missing something. If so, I'd appreciate being corrected.

**Comment by stygstra** [comment]
[Another place](https://github.com/farizrahman4u/keras-contrib/blob/c4d1a617461b0a4e29ec087de562032d47d03244/keras_contrib/layers/normalization.py#L293-L299) where `K.get_value` is used:
```python
        r_max_value = K.get_value(self.r_max)
        r = std_batch / (K.sqrt(self.running_variance + self.epsilon))
        r = K.stop_gradient(K.clip(r, 1 / r_max_value, r_max_value))

        d_max_value = K.get_value(self.d_max)
        d = (mean_batch - self.running_mean) / K.sqrt(self.running_variance + self.epsilon)
        d = K.stop_gradient(K.clip(d, -d_max_value, d_max_value))
```

Since this is in call(), the clipping is fixed at `r_max_value=1` and `d_max_value=0` rather than smoothly approaching `r_max_value=3` and `d_max_value=5` as described in section 4, paragraph 3 of the [batch renorm paper](https://arxiv.org/pdf/1702.03275.pdf).

The fix should be pretty easy. #128 and #142 make it so that `self.r_max` and `self.d_max` get updated every evaluation, so all that needs to be done is replace the above with:
```python
        r = std_batch / (K.sqrt(self.running_variance + self.epsilon))
        r = K.stop_gradient(K.clip(r, 1 / self.r_max, self.r_max))

        d = (mean_batch - self.running_mean) / K.sqrt(self.running_variance + self.epsilon)
        d = K.stop_gradient(K.clip(d, -self.d_max, self.d_max))
```

**Comment by stygstra** [comment]
Related observation: with the current parameters, `self.r_max` and `self.d_max` converge much more quickly to `self.r_max_value` and `self.d_max_value` than the [batch renorm paper](https://arxiv.org/pdf/1702.03275.pdf) suggests in section 4, paragraph 3. Once this bug (#127) is fixed, this will need to be considered.

The paper warns:
> However, at the beginning of training, when the learning rate was larger, it proved important to increase r_max slowly: otherwise, occasional large gradients were observed to suddenly and severely increase the loss.

The paper suggests:
* fixing `r_max=1` and `d_max=0` for the first 5000 steps,
* gradually increasing `r_max` from 1 to 3 @ 40k steps, and
* gradually increasing `d_max` from 0 to 5 @ 25k steps.

By default, BatchRenormalization sets `t_delta=1e-3` and uses the update rules:
```python
        r_val = self.r_max_value / (1 + (self.r_max_value - 1) * K.exp(-self.t))
        d_val = self.d_max_value / (1 + ((self.d_max_value / 1e-3) - 1) * K.exp(-(2 * self.t)))
```

These update rules immediately begin increasing `r_max` and reach `r_max=3` and `d_max=5` after only around 6k steps:
![plot of r_max,d_max](https://user-images.githubusercontent.com/203361/29861477-3c96886a-8d1e-11e7-9589-0a9716cca249.png)
```python
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
sns.set()

steps = np.arange(10000)
t_delta = 1e-3
t = steps*t_delta
r_max = 3/(1+(3-1)*np.exp(-t))
d_max = 5/(1+(5/1e-3 - 1)*np.exp(-2*t))

plt.plot(steps, r_max, label='r_max')
plt.plot(steps, d_max, label='d_max')
plt.legend()
plt.show()
```
